src/test/java/org/java_websocket/issues/Issue900Test.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+package org.java_websocket.issues;
+
+import org.java_websocket.WebSocket;
+import org.java_websocket.WebSocketImpl;
+import org.java_websocket.WrappedByteChannel;
+import org.java_websocket.client.WebSocketClient;
+import org.java_websocket.handshake.ClientHandshake;
+import org.java_websocket.handshake.ServerHandshake;
+import org.java_websocket.server.WebSocketServer;
+import org.java_websocket.util.SocketUtil;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.concurrent.CountDownLatch;
+
+public class Issue900Test {
+
+    CountDownLatch countServerDownLatch = new CountDownLatch(1);
+    CountDownLatch countCloseCallsDownLatch = new CountDownLatch(1);
+
+    @Test(timeout = 2000)
+    public void testIssue() throws Exception {
+        int port = SocketUtil.getAvailablePort();
+        final WebSocketClient client = new WebSocketClient(new URI("ws://localhost:" + port)) {
+            @Override
+            public void onOpen(ServerHandshake handshakedata) {
+
+            }
+
+            @Override
+            public void onMessage(String message) {
+            }
+
+            @Override
+            public void onClose(int code, String reason, boolean remote) {
+            }
+
+            @Override
+            public void onError(Exception ex) {
+
+            }
+        };
+        WebSocketServer server = new WebSocketServer(new InetSocketAddress(port)) {
+            @Override
+            public void onOpen(WebSocket conn, ClientHandshake handshake) {
+            }
+
+            @Override
+            public void onClose(WebSocket conn, int code, String reason, boolean remote) {
+                countCloseCallsDownLatch.countDown();
+            }
+
+            @Override
+            public void onMessage(WebSocket conn, String message) {
+
+            }
+
+            @Override
+            public void onError(WebSocket conn, Exception ex) {
+
+            }
+
+            @Override
+            public void onStart() {
+                countServerDownLatch.countDown();
+            }
+        };
+        new Thread(server).start();
+        countServerDownLatch.await();
+        client.connectBlocking();
+        WebSocketImpl websocketImpl = (WebSocketImpl)new ArrayList<WebSocket>(server.getConnections()).get(0);
```

Comment:
Is there a reason for not writing it like this:
```java
WebSocketImpl websocketImpl = (WebSocketImpl) server.getConnections().get(0)
```

Reply:
getConnections() returns a collection